src/authentication/MobileRouter.jsx

```diff
@@ -1,39 +1,119 @@
-import React from 'react'
+import React, { Component } from 'react'
 import { Router, withRouter } from 'react-router'
-
+import Proptypes from 'prop-types'
 import Authentication from './src/Authentication'
 import Revoked from './src/Revoked'
 import { logException } from 'drive/lib/reporter'
+import {
+  readState,
+  secretExchange,
+  getAccessToken,
+  readSecret,
+  clearSecret,
+  clearState,
+  checkExchangedInformations,
+  checkIfOnboardingLogin
+} from './src/utils/onboarding'
+
+class MobileRouter extends Component {
+  async checkState(receivedState, code, cozy_url, history) {
```

Comment:
What is `receivedState `? We're not using it at all as far as I can tell.

Reply:
Oh, good catch! We need to check the receivedState with our localState before calling the `secretExchange`